### PR TITLE
Class decorator to attach instances to the window object

### PR DESCRIPTION
**Pull request by bahmutov**
- [x] class decorators are working
- [x] tests are passing
- [x] update root readme
- [x] write recipe README

Main idea: each component class uses a class decorator like this
```js
import { CypressInstances } from '../decorators'

@CypressInstances()
export default class TodoItem extends Component {
```

This decorator when running inside Cypress test automatically saves singleton on the `window` object, allowing the test to easily reach inside the app and directly check state or invoke methods.

```js
it('creates Todo components', () => {
  cy.visit('/')
  cy.get('.new-todo').type('first{enter}').type('second{enter}').type('third{enter}')

  cy.window().its('TodoItem').should('have.length', 3)
  // let's complete second todo
  .its('1.props').then((props) => {
    props.completeTodo(props.todo.id)
  })

  // UI updates
  cy.get('li.todo').eq(1).should('have.class', 'completed')
})
```

### CODE REVIEW

File: examples/blogs__app-decorator/README.md [hunk 1]
```diff
@@ -0,0 +1,38 @@
+# App decorator
+> Using class decorator to automatically attach instance to window
+
+The application source code uses a [class decorator](https://www.sitepoint.com/javascript-decorators-what-they-are/) to automatically save created instances on the `window` object when running inside Cypress.
```
Comment (jennifer-shehane) [inline examples/blogs__app-decorator/README.md:4]: Some inconsistency on the naming - app decorators vs class decorators. I think they're just called 'decorators' or 'JavaScript decorators' typically. I'd stick to the most used terminology for better find-ability. 

I don't see the use of 'App decorators' in google search at all and a few mentions of 'class decorators' although those span across languages outside of JavaScript.

File: examples/blogs__app-decorator/README.md [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+    })
+
+  // UI updates
+  cy.get('li.todo').eq(1).should('have.class', 'completed')
+})
+```
+
+![Test](images/app-decorator.png)
+
+Note: class decorators are an experimental JavaScript feature. In this recipe they are transpiled using Babel plugins in [.babelrc](.babelrc) file.
```
Comment (jennifer-shehane) [inline examples/blogs__app-decorator/README.md:38]:
```suggestion
Note: class decorators are an experimental JavaScript feature. In this recipe they are transpiled using Babel plugins in the [.babelrc](.babelrc) file.
```

File: examples/blogs__app-decorator/README.md (hunk 1 above)
Comment (bahmutov) [reply examples/blogs__app-decorator/README.md:4]: Yeah, it is kind of mixing: so this is class decorator that allows the user to expose an instance to do "app action". Let me just use JavaScript / class decorator for consistency.

Comment (bahmutov) [reply examples/blogs__app-decorator/README.md:4]: I will rename the recipe

**Pull request merged**